### PR TITLE
S08 t06 all fixes

### PR DESCRIPTION
**Pull request by FlorenceBuchelet**

### CODE REVIEW

**Review by VincentRssx — APPROVED** [review]

File: frontend/src/components/DecisionCard/DecisionCard.jsx [hunk 1]
```diff
@@ -20,7 +20,8 @@ function DecisionCard({
   return (
     <Link onClick={handleClick} to="/decision">
       <button type="button" className="decisionCard__container">
-        <h2>{title}</h2>
+        {/* Titles are too long for the small cards in homepage, only displaying the first 50 characeters */}
```
Comment (fantinerudent) [inline frontend/src/components/DecisionCard/DecisionCard.jsx:23]: characters

File: frontend/src/pages/CreateDecision/CreateDecision.jsx [hunk 2]
```diff
@@ -22,6 +22,13 @@ function CreateDecision() {
   // Hook pour la navigation
   const navigate = useNavigate();
 
+  // Le site n'est accessible qu'en étant connecté
```
Comment (fantinerudent) [inline frontend/src/pages/CreateDecision/CreateDecision.jsx:25]: if you let comments in your code, at least let them in English :)

File: frontend/src/pages/CreateDecision/CreateDecision.jsx [hunk 3]
```diff
@@ -53,7 +60,7 @@ function CreateDecision() {
 
       // Redirection vers la page de connexion si la création réussit
```
Comment (fantinerudent) [inline frontend/src/pages/CreateDecision/CreateDecision.jsx:61]: english

File: frontend/src/pages/CreateDecision/CreateDecision.jsx [hunk 4]
```diff
@@ -53,7 +60,7 @@ function CreateDecision() {
...
       if (response.status === 201) {
-        navigate("/decision");
+        navigate("/homepage");
       } else {
         // Log des détails de la réponse en cas d'échec
```
Comment (fantinerudent) [inline frontend/src/pages/CreateDecision/CreateDecision.jsx:65]: english

File: frontend/src/pages/Homepage/Homepage.jsx [hunk 5]
```diff
@@ -1,10 +1,23 @@
...
 import DecisionCard from "../../components/DecisionCard/DecisionCard";
+import { AuthContext } from "../../contexts/authContext";
 
 function Homepage() {
   const [decisions, setDecisions] = useState();
+
+  const navigate = useNavigate();
+  const { user } = useContext(AuthContext);
+
+  // Le site n'est accessible qu'en étant connecté
```
Comment (fantinerudent) [inline frontend/src/pages/Homepage/Homepage.jsx:14]: english

File: frontend/src/pages/Decision/Decision.jsx [hunk 6]
```diff
@@ -29,6 +33,16 @@ function Decision() {
     status: "--",
     user_id: 0,
   });
+
+  const navigate = useNavigate();
+
+  // Le site n'est accessible qu'en étant connecté
```
Comment (fantinerudent) [inline frontend/src/pages/Decision/Decision.jsx:39]: english

File: frontend/src/pages/CreateUser/CreateUser.jsx [hunk 7]
```diff
@@ -10,6 +12,18 @@ function CreateUser() {
   const [isAdmin, setIsAdmin] = useState(false);
   const [message, setMessage] = useState("");
 
+  const navigate = useNavigate();
+  const { user } = useContext(AuthContext);
+
+  // Cette page n'est accessible qu'aux admins
```
Comment (fantinerudent) [inline frontend/src/pages/CreateUser/CreateUser.jsx:18]: english

**Review by fantinerudent — APPROVED** [review]

**Pull request merged**